slasher/beaconclient/validator_retrieval.go

```diff
@@ -0,0 +1,43 @@
+package beaconclient
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"go.opencensus.io/trace"
+)
+
+// RequestValidator requests validator public key from a beacon node via gRPC.
+func (bs *Service) RequestValidator(
+	ctx context.Context,
+	validatorIdx uint64,
```

Comment:
Also, this function should be modeled to batch responses. We should need to ping the beacon node for every N validators on an attestation.

Reply:
replaced with ListValidators endpoint